java/vector/src/main/java/org/apache/arrow/vector/complex/ListViewVector.java

```diff
@@ -0,0 +1,913 @@
+   */
+  @Override
+  public boolean isEmpty(int index) {
+    if (isNull(index)) {
+      return true;
+    } else {
+      return sizeBuffer.getInt(index * SIZE_WIDTH) == 0;
+    }
+  }
+
+  /**
+   * Same as {@link #isNull(int)}.
+   *
+   * @param index  position of the element
+   * @return 1 if element at given index is not null, 0 otherwise
+   */
+  public int isSet(int index) {
+    final int byteIndex = index >> 3;
+    final byte b = validityBuffer.getByte(byteIndex);
+    final int bitIndex = index & 7;
+    return (b >> bitIndex) & 0x01;
+  }
+
+  /**
+   * Get the number of elements that are null in the vector.
+   *
+   * @return the number of null elements.
+   */
+  @Override
+  public int getNullCount() {
+    return BitVectorHelper.getNullCount(validityBuffer, valueCount);
+  }
+
+  /**
+   * Get the value capacity by considering validity and offset capacity.
+   * Note that the size buffer capacity is not considered here since it has
+   * the same capacity as the offset buffer.
+   *
+   * @return the value capacity
+   */
+  @Override
+  public int getValueCapacity() {
+    return getValidityAndOffsetValueCapacity();
+  }
+
+  private int getValidityAndSizeValueCapacity() {
+    final int offsetValueCapacity = Math.max(getOffsetBufferValueCapacity(), 0);
+    final int sizeValueCapacity = Math.max(getSizeBufferValueCapacity(), 0);
+    return Math.min(offsetValueCapacity, sizeValueCapacity);
+  }
+
+  private int getValidityAndOffsetValueCapacity() {
+    final int offsetValueCapacity = Math.max(getOffsetBufferValueCapacity(), 0);
+    return Math.min(offsetValueCapacity, getValidityBufferValueCapacity());
+  }
+
+  private int getValidityBufferValueCapacity() {
+    return capAtMaxInt(validityBuffer.capacity() * 8);
+  }
+
+  /**
+   * Set the element at the given index to null.
+   * @param index the value to change
+   */
+  @Override
+  public void setNull(int index) {
+    while (index >= getValidityAndSizeValueCapacity()) {
+      reallocValidityAndSizeAndOffsetBuffers();
+    }
+    if (lastSet >= index) {
+      lastSet = index - 1;
+    }
+
+    if (index == 0) {
+      offsetBuffer.setInt(0, 0);
+      sizeBuffer.setInt(0, 0);
+    } else {
+      final int prevOffset = offsetBuffer.getInt((index - 1) * OFFSET_WIDTH);
+      final int prevSize = sizeBuffer.getInt((index - 1) * SIZE_WIDTH);
+      final int currOffSet = prevOffset + prevSize;
+      offsetBuffer.setInt(index * OFFSET_WIDTH, currOffSet);
+      sizeBuffer.setInt(index * SIZE_WIDTH, 0);
+    }
+
+    BitVectorHelper.unsetBit(validityBuffer, index);
+    lastSet = index;
+  }
+
+  /**
+   * Start new value in the ListView vector.
+   * There are a few cases that are handled in this function.
+   * There are two main scenarios that need to be considered.
+   * The first scenario is simple insertion where indices are continuously updated.
+   * The other scenario is the event of non-continuous writing,
+   * the offset buffer needs to be updated.
```

Comment:
This sort of discussion doesn't belong in the docstring...

Reply:
removed